### PR TITLE
Fix logging to stdout

### PR DESCRIPTION
**Pull request by schans**
Allow for setting logFile to /dev/stdout for logging to stdout. Log settings are set directly in the mongodb-agent comment instead of passing through the environment (which is not the most secure way).

### All Submissions:

* [X] Have you opened an Issue before filing this PR?
* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [X] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

**Comment by schans** [comment]
This addresses https://github.com/mongodb/mongodb-kubernetes-operator/issues/1439

**Comment by slaskawi** [comment]
Let me assign this to @lsierant for review

**Comment by schans** [comment]
Thanks!

**Comment by schans** [comment]
Hi @lsierant ,

Please don't hesitate to reach out to me if you need some extra info or if you would like some update to the PR. I'm happy to work with you to find a solution for this as it is blocking our upgrades to newer mongo versions.

Thanks
Bart

**Comment by schans** [comment]
Updated unit test. Instead of testing for the env variables it now test for the construction of the agent command.

**Comment by schans** [comment]
Sorry. Missed the env count check, should be all green now.

**Comment by schans** [comment]
Hi @slaskawi and @lsierant, 

Thank you for all your support so far. Please let me know how I can help to get this merged/fixed!

File: controllers/construct/mongodbstatefulset.go [hunk 1]
```diff
@@ -259,11 +255,23 @@ func BaseAgentCommand() string {
...
 // it can be used to register the agent with OM.
-func AutomationAgentCommand(withAgentAPIKeyExport bool) []string {
+func AutomationAgentCommand(withAgentAPIKeyExport bool, logLevel string, logFile string, maxLogFileDurationHours int) []string {
+	// This is somewhat undocumented at https://www.mongodb.com/docs/ops-manager/current/reference/mongodb-agent-settings/
+	// Not setting the -logFile option make the mongodb-agent log to stdout. Setting -logFile /dev/stdout will result in
+	// an error by the agent trying to open /dev/stdout-verbose and still trying to do log rotation.
+	// To keep consistent with old behavior not setting the logFile in the config does not log to stdout but keeps
+	// the default logFile as defined by DefaultAgentLogFile. Setting the logFile explictly to "/dev/stdout" will log to stdout.
+	agentLogOptions := ""
+	if logFile != "/dev/stdout" {
```
Comment (nammn) [inline controllers/construct/mongodbstatefulset.go:265]: From a reading perspective I prefer the inverse:

```
if logFile == ...{}

else:
{}
```

**Review by nammn — APPROVED** [review]
LGTM and thanks for your contribution!

**Comment by legal90** [comment]
@schans Thank you for fixing this issue! 👍 

@nammn Do you have any estimates when there could be a new version of community operator with this fix included? Thanks!

**Comment by jlestrange** [comment]
Hi @nammn 

Any update on when this is going to be released, we need this update to be able to use the latest version of the app?

Thanks

**Comment by nammn** [comment]
@jlestrange sorry for the late release, but its out now!
https://github.com/mongodb/mongodb-kubernetes-operator/releases/tag/v0.10.0

**Comment by nammn** [comment]
@jlestrange @uttammeena  @legal90  @laiminhtrung1997 sorry for the late reply and release, but its out now!
https://github.com/mongodb/mongodb-kubernetes-operator/releases/tag/v0.10.0

**Comment by laiminhtrung1997** [comment]
Dear @nammn @schans 
I’m currently utilizing MongoDB Operator version 0.9.0 and have configured the agent.logFile to /dev/stdout. As per my understanding from issue #1439, all log levels will now be directed to stdout instead of just agent.logLevel: INFO, correct? If that’s the case, wouldn’t this lead to the occurrence of issue #1226? Could you please help me confirm this?

**Comment by nammn** [comment]
Hey @laiminhtrung1997
- if configured to stdout all loglevels will go to stdout
- no, the volume shouldn't fill up - since all logs go to stdout and not to a file, can you clarify what you mean?

**Comment by laiminhtrung1997** [comment]
Dear @nammn 
Sorry for my silly question, I want to ensure the issue from #1226 has not occurred in my environment. Thanks for your quick reply.

**Pull request merged**